tests/schema/test_interface.py

```diff
@@ -93,3 +94,35 @@ def always_error(self) -> Error:
         "field": "Password",
         "fix": "Choose more characters",
     }
+
+
+def test_interface_duck_typing():
+    @strawberry.interface
+    class Entity:
+        id: int
+
+    @strawberry.type
+    class Anime(Entity):
+        name: str
+
+    @dataclass
+    class AnimeORM:
+        id: int
+        name: str
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def anime(self) -> Anime:
+            return AnimeORM(id=1, name="One Piece")  # type: ignore
```

Comment:
Are we sure we want to support this kinda of duck typing? This feels like potential for bugs in user code.

Reply:
IMO we shouldn't support it. I think it leads to more issues than it solves. It's related to this discussion: https://github.com/strawberry-graphql/strawberry/discussions/515

Reply:
do we want to support returning dictionaries though?

Reply:
I see strawberry types as protocols, we should be able to allow users to returns objects (and dictionary) that match the shape :)

We can also add options to disable this behaviour though (maybe enabled by default)

Reply:
> I see strawberry types as protocols, we should be able to allow users to returns objects (and dictionary) that match the shape :)

I like the idea but IMO it doesn't fit with the ability to define resolver functions on the type directly or with the dataclasses API generally.

If the Strawberry API was something like this:

```python
MyType = strawberry.type(name='MyType', fields={
	"name": str,
	"age": strawberry.field(str, resolve=lambda obj: 20),
})
```

then it makes sense to allow resolvers to return dictionaries (this is basically the graphql-core API)

The conclusion we came to in https://github.com/strawberry-graphql/strawberry/discussions/515 (which I still agree with) is that we don't disallow returning other things than StrawberryTypes from a resolver, but it's only an advanced option and the docs should always return StrawberryTypes.

Reply:
what about:

```python
@strawberry.type
class MyType:
    age: strawberry.field(resolver=some_resolver)
```

?

> The conclusion we came to in #515 (which I still agree with) is that we don't disallow returning other things than StrawberryTypes from a resolver, but it's only an advanced option and the docs should always return StrawberryTypes.

I think we should write it somewhere that we support this (either by default or with an option) :)

Reply:
> what about:
>
> @strawberry.type
> class MyType:
>     age: strawberry.field(resolver=some_resolver)

What are you suggesting here? That that is the only way to define resolvers?

> I think we should write it somewhere that we support this (either by default or with an option) :)

Yep totally agree.

Reply:
> What are you suggesting here? That that is the only way to define resolvers?

No, I was wondering if that does fall into the "issue" mentioned here https://github.com/strawberry-graphql/strawberry/discussions/515

Reply:
Well the issue is what should the call signature of `some_resolver` look like?

It could be:

```python
def some_resolver(obj: MyType) -> int
```

or it could be:

```python
def some_resolver(obj: dict) -> int
```

it is completely dependant on what the resolver for `MyType` returns in the parent type. If we support returning whatever you want from resolvers that return object types then your child resolvers need to take into account all the possible options. If we say that object type resolvers only return an instance of the type, then the resolver signature can only look like the first option. IMO it reduces a whole bunch of confusion.

Reply:
right, I didn't consider that, though it would be ok for Protocols/duck-typing. Not for dicts. I wonder if we could add some magic, but let's leave for the future :D